engine/src/graphics/device.cpp

```diff
@@ -15,16 +15,18 @@
 namespace gen
 {
 
-	GraphicsDevice::GraphicsDevice(std::string const & appName)
+	GraphicsDevice::GraphicsDevice(Window &window, std::string const & appName)
 	{
 		// TODO: Allow for the vulkan api to be specified in a config file or through command line arguments.
 		// We could also dynamically check for the highest supported version of the vulkan api, but that feels outside the scope of this project.
 		createInstance(appName, "Genesis Engine", VK_API_VERSION_1_3);
+		createSurface(window);
 	}
 
 	GraphicsDevice::~GraphicsDevice()
 	{
 
+		m_surface.release();
```

Comment:
Why are we doing this manually though? Just having a default destructor and the correct member order will take care of it automatically. Kinda defeats the purpose of using RAII types.